llvm/test/SYCLLowerIR/printf_addrspace/diagnose_bad_wrapper_inlining.ll

```diff
@@ -0,0 +1,47 @@
+;; This tests diagnostic emission whenever printf's format string argument
+;; is taken from the argument list of a user wrapper function, but then the
+;; user-provided wrapper does not get inlined (while the library-provided
+;; experimental::printf wrapper does).
+
+;; The IR for test purposes is based on the following source/compilation (custom
+;; build of SYCL Clang with SYCLMutatePrintfAddrspacePass turned off):
+;; clang++ -fsycl -fsycl-device-only Inputs/experimental-printf-bad-inline-test.cpp -S -D__SYCL_USE_NON_VARIADIC_SPIRV_OCL_PRINTF__ -O2
+
+; RUN: not opt < %s --SYCLMutatePrintfAddrspace -S 2>&1 | FileCheck %s
+; RUN: not opt < %s --SYCLMutatePrintfAddrspace -S --enable-new-pm=1 2>&1 | FileCheck %s
```

Comment:
```suggestion
; RUN: not opt < %s --SYCLMutatePrintfAddrspace -S --enable-new-pm=0 2>&1 | FileCheck %s
; RUN: not opt < %s -passes=SYCLMutatePrintfAddrspace -S 2>&1 | FileCheck %s
```

Reply:
Done in https://github.com/intel/llvm/pull/5306/commits/57cec9b1a9877dc46e762b5ec2851b81f5f831ed for all of the new tests.